src/mlpack/core/optimizers/cmaes/test.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file test.cpp
+ * Very short example of CMAES optimizer working
+ */
+
+#include <stdlib.h>
+#include <iostream>
+#include <math.h>
+
+#include "cmaes.hpp"
+
+using namespace mlpack;
+using namespace mlpack::optimization;
+/**
+* fitness function defined
+*/
+double fitfun(double const *x, int N)
+{
+  double func = (pow(x[0]-5, 2))*(pow(x[1]-3, 2)) ; // function = (x-5)^2 * (y-3)^2 (minima at x=5 y=3)
+  return func;
+}
+
+int main(int, char**)
+ {
+
+  CMAES<double> evo;
```

Comment:
We should pass the function we like to evaluate and do all the performance evaluation inside the optimizer, just like we did for the other optimizer classes. Also we don't have to create a new function for the test, we could reuse the sgd test function:

```c++
SGDTestFunction f;
CMAES<SGDTestFunction> s(f);

arma::mat coordinates = f.GetInitialPoint();
double result = s.Optimize(coordinates);
```

in this case ``coordinates`` is the parameter we like to optimize, in your case this is ``x``. If you need help with the transition, let me know.